docs/get-started/how-to/state-recovery.mdx

```diff
@@ -0,0 +1,104 @@
+
+To this end, the Linea Besu execution client has a state recovery plugin that:
+
+1. Retrieves finalized blobs from L1; 
+2. Decompresses the blob data;
+3. Replays the transactions in the execution client, and, finally,
+4. Checks that the resulting `stateRootHash` matches the one finalized on L1. 
+
+Linea Besu executes the fetched transactions in the blob, while the state management plugin, Shomei, 
+rebuilds state.
+
+## Prerequisites
+
+:::note Availability
+
+The state recovery plugin only works from the following block numbers onwards:
+- Linea Mainnet: TBC (once Beta v2 is live)
+- Linea Sepolia: `9572301` (the block where the 2B gas limit was implemented)
+
+:::
+
+The Linea Besu node being used must have been running with the following configuration in the `.toml` 
+file since genesis (i.e. from when it started to sync):
+- [`FULL`](https://besu.hyperledger.org/public-networks/concepts/node-sync#full-synchronization) sync
+mode:
+    ```toml
+    sync-mode="FULL"
+    ```
+- [`BONSAI`](https://besu.hyperledger.org/public-networks/reference/cli/options#data-storage-format) 
+data storage:
+    ```toml
+    data-storage-format="BONSAI"
+    ```
+
+
+To learn how to run a Linea Besu node, see our [guide](../how-to/run-a-node/linea-besu.mdx).
+
+## Update configuration
+
+To proceed with state recovery, your node must have been running normally, in line with the above 
+conditions (i.e. `FULL` sync mode and `BONSAI` data storage enabled).
+
+:::info 
+
+Only enable state recovery if you intend the node to be used specifically for state recovery. Even 
+if you do not specify a block from which state recovery should start, default behavior will 
+ensure that state recovery will replace standard P2P syncing. It's therefore not possible to add
+the state recovery plugin and configuration without it affecting your node's behavior.
+
+:::
+
+Only then can you proceed by adding the following configuration parameters.
+
+Insert them into the `.toml` configuration file alongside any existing plugin configurations, or via
+the command line by adding `--` before each parameter:
+
+```toml
+### Plugin Configuration ###
+Xplugins-external-enabled=true
+Xbonsai-limit-trie-logs-enabled=false
+plugins=["BesuShomeiRpcPlugin","ZkTrieLogPlugin","LineaStateRecoveryPlugin"]
+plugin-shomei-http-host="11.11.13.11"
+plugin-shomei-http-port=8888
+plugin-staterecovery-linea-sequencer-beneficiary-address="0xa27342f1b74c0cfb2cda74bac1628d0c1a9752f2"
+plugin-staterecovery-l1-smart-contract-address="0xb218f8a4bc926cf1ca7b3423c154a0d627bdb7e5"
+plugin-staterecovery-l1-endpoint="https://sepolia.infura.io/v3/INFURA_APP_KEY"
+plugin-staterecovery-l1-polling-interval="PT12S"
+plugin-staterecovery-l1-success-backoff-delay="PT2S"
+plugin-staterecovery-l1-retry-backoff-delay="PT4S"
+plugin-staterecovery-l1-earliest-search-block="7316705"
+plugin-staterecovery-l1-highest-search-block="FINALIZED"
+plugin-staterecovery-shomei-endpoint="http://shomei-sr-sepolia:8888"
```

Comment:
this may be missleading as `shomei-sr-sepolia` does not exist. 

Additionally, to verify the state we need to run 2 components: 
- Besu with 2 plugins: Shomei and Staterecovery
- Shomei : the actual Linea type 2 statemanager 

This documentation shall be extended with Shomei instructions and configuration example to match oneanother.

Reply:
So we need to run Shomei separately as well, alongside Linea Besu? 

> Besu with 2 plugins: Shomei and Staterecovery

Have we addressed these two components in the existing instructions?